src/expr/src/scalar/func.rs

```diff
@@ -2198,6 +2198,9 @@ fn range_difference<'a>(
     propagates_nulls = true
 )]
 fn eq<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
+    // SQL equality demands that if either input is null, then the result should be null. However,
+    // we don't need to handle this case here; it is handled when `BinaryFunc::eval` checks
+    // `propagates_nulls`.
```

Comment:
Various people were repeatedly baffled by not seeing where the null case is implemented, see e.g. https://www.notion.so/materialize/Optimizer-discussions-in-Slack-85940c05f68148828a33629b063def10?source=copy_link#9c681b63c5184ff5803b7a837c83930c